src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,106 @@
+package mate.academy.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.BookDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private final Connection connection;
+
+    public BookDaoImpl(Connection connection) {
+        this.connection = connection;
+    }
+
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (PreparedStatement statement = connection.prepareStatement(query,
+                Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            if (statement.executeUpdate() < 1) {
+                throw new RuntimeException("Couldn't create book " + book);
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
```

Comment:
```suggestion
                Long id = generatedKeys.getObject(1, Long.class);
                book.setId(id);
```

see common mistakes file